[CI] TSDBIndexingIT testDynamicDimensions failing

### CI Link

https://gradle-enterprise.elastic.co/s/w5r7w6kvfn24a/tests/task/:modules:data-streams:internalClusterTest/details/org.elasticsearch.datastreams.TSDBIndexingIT/testDynamicDimensions?top-execution=1

### Repro line

`./gradlew ":modules:data-streams:internalClusterTest" --tests "org.elasticsearch.datastreams.TSDBIndexingIT.testDynamicDimensions" -Dtests.seed=C529FD384F25507E -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=mer -Dtests.timezone=Atlantic/South_Georgia -Druntime.java=25`

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

    java.lang.AssertionError: expected [index.mapping.use_synthetic_id] to be private but it was not	
	at __randomizedtesting.SeedInfo.seed([C529FD384F25507E]:0)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.validatePrivateSettingsNotExplicitlySet(MetadataCreateIndexService.java:1631)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.getIndexSettingsValidationErrors(MetadataCreateIndexService.java:1617)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:2185)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:2101)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validate(MetadataIndexTemplateService.java:2096)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.validateIndexTemplateV2(MetadataIndexTemplateService.java:834)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:740)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.addIndexTemplateV2(MetadataIndexTemplateService.java:676)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$5.execute(MetadataIndexTemplateService.java:608)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$TemplateClusterStateUpdateTask.execute(MetadataIndexTemplateService.java:176)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:152)	
	at org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:149)	
	at org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:71)

    WARNING: Suite execution timed out: org.elasticsearch.datastreams.TSDBIndexingIT

--- a/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
+++ b/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
@@ -596,9 +596,7 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName, reindexedDataStreamName))
-                .template(
-                    new Template(templateSettings.build(), new CompressedXContent(MAPPING_TEMPLATE), null)
-                )
+                .template(new Template(templateSettings.build(), new CompressedXContent(MAPPING_TEMPLATE), null))
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
                 .build()
         );
@@ -741,37 +739,31 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName))
-                .template(
-                    new Template(
-                        templateSettings.build(),
-                        new CompressedXContent("""
-                                {
-                              "_doc": {
-                                "dynamic_templates": [
-                                  {
-                                    "labels": {
-                                      "match_mapping_type": "string",
-                                      "mapping": {
-                                        "type": "keyword",
-                                        "time_series_dimension": true
-                                      }
-                                    }
-                                  }
-                                ],
-                                "properties": {
-                                  "@timestamp": {
-                                    "type": "date"
-                                  },
-                                  "metricset": {
-                                    "type": "keyword",
-                                    "time_series_dimension": true
-                                  }
-                                }
+                .template(new Template(templateSettings.build(), new CompressedXContent("""
+                        {
+                      "_doc": {
+                        "dynamic_templates": [
+                          {
+                            "labels": {
+                              "match_mapping_type": "string",
+                              "mapping": {
+                                "type": "keyword",
+                                "time_series_dimension": true
                               }
-                            }"""),
-                        null
-                    )
-                )
+                            }
+                          }
+                        ],
+                        "properties": {
+                          "@timestamp": {
+                            "type": "date"
+                          },
+                          "metricset": {
+                            "type": "keyword",
+                            "time_series_dimension": true
+                          }
+                        }
+                      }
+                    }"""), null))
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
                 .build()
         );
@@ -813,37 +805,31 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName))
-                .template(
-                    new Template(
-                        templateSettings.build(),
-                        new CompressedXContent("""
+                .template(new Template(templateSettings.build(), new CompressedXContent("""
 
-                                {
-                              "_doc": {
-                                "dynamic_templates": [
-                                  {
-                                    "label": {
-                                      "mapping": {
-                                        "type": "keyword",
-                                        "time_series_dimension": true
-                                      }
-                                    }
-                                  }
-                                ],
-                                "properties": {
-                                  "@timestamp": {
-                                    "type": "date"
-                                  },
-                                  "metricset": {
-                                    "type": "keyword",
-                                    "time_series_dimension": true
-                                  }
-                                }
+                        {
+                      "_doc": {
+                        "dynamic_templates": [
+                          {
+                            "label": {
+                              "mapping": {
+                                "type": "keyword",
+                                "time_series_dimension": true
                               }
-                            }"""),
-                        null
-                    )
-                )
+                            }
+                          }
+                        ],
+                        "properties": {
+                          "@timestamp": {
+                            "type": "date"
+                          },
+                          "metricset": {
+                            "type": "keyword",
+                            "time_series_dimension": true
+                          }
+                        }
+                      }
+                    }"""), null))
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
                 .build()
         );

--- a/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
+++ b/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
@@ -588,19 +588,16 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
     public void testReindexing() throws Exception {
         String dataStreamName = "my-ds";
         String reindexedDataStreamName = "my-reindexed-ds";
+        var templateSettings = Settings.builder().put("index.mode", "time_series");
+        if (IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean()) {
+            templateSettings.put(IndexSettings.USE_SYNTHETIC_ID.getKey(), true);
+        }
         var putTemplateRequest = new TransportPutComposableIndexTemplateAction.Request("id");
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName, reindexedDataStreamName))
                 .template(
-                    new Template(
-                        Settings.builder()
-                            .put("index.mode", "time_series")
-                            .put(IndexSettings.USE_SYNTHETIC_ID.getKey(), IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean())
-                            .build(),
-                        new CompressedXContent(MAPPING_TEMPLATE),
-                        null
-                    )
+                    new Template(templateSettings.build(), new CompressedXContent(MAPPING_TEMPLATE), null)
                 )
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
                 .build()
@@ -650,20 +647,16 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
         String dataStreamName = "my-ds";
         var putTemplateRequest = new TransportPutComposableIndexTemplateAction.Request("id");
         boolean indexDimensionsTsidStrategyEnabled = randomBoolean();
+        var templateSettings = Settings.builder()
+            .put("index.mode", "time_series")
+            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabled);
+        if (IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean()) {
+            templateSettings.put(IndexSettings.USE_SYNTHETIC_ID.getKey(), true);
+        }
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName))
-                .template(
-                    new Template(
-                        Settings.builder()
-                            .put("index.mode", "time_series")
-                            .put("index.dimensions_tsid_strategy_enabled", indexDimensionsTsidStrategyEnabled)
-                            .put(IndexSettings.USE_SYNTHETIC_ID.getKey(), IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean())
-                            .build(),
-                        new CompressedXContent(MAPPING_TEMPLATE),
-                        null
-                    )
-                )
+                .template(new Template(templateSettings.build(), new CompressedXContent(MAPPING_TEMPLATE), null))
                 .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
                 .build()
         );
@@ -740,16 +733,17 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
 
     public void testDynamicStringDimensions() throws Exception {
         String dataStreamName = "my-ds";
+        var templateSettings = Settings.builder().put("index.mode", "time_series");
+        if (IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean()) {
+            templateSettings.put(IndexSettings.USE_SYNTHETIC_ID.getKey(), true);
+        }
         var putTemplateRequest = new TransportPutComposableIndexTemplateAction.Request("id");
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName))
                 .template(
                     new Template(
-                        Settings.builder()
-                            .put("index.mode", "time_series")
-                            .put(IndexSettings.USE_SYNTHETIC_ID.getKey(), IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean())
-                            .build(),
+                        templateSettings.build(),
                         new CompressedXContent("""
                                 {
                               "_doc": {
@@ -811,16 +805,17 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
 
     public void testDynamicDimensions() throws Exception {
         String dataStreamName = "my-ds";
+        var templateSettings = Settings.builder().put("index.mode", "time_series");
+        if (IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean()) {
+            templateSettings.put(IndexSettings.USE_SYNTHETIC_ID.getKey(), true);
+        }
         var putTemplateRequest = new TransportPutComposableIndexTemplateAction.Request("id");
         putTemplateRequest.indexTemplate(
             ComposableIndexTemplate.builder()
                 .indexPatterns(List.of(dataStreamName))
                 .template(
                     new Template(
-                        Settings.builder()
-                            .put("index.mode", "time_series")
-                            .put(IndexSettings.USE_SYNTHETIC_ID.getKey(), IndexSettings.TSDB_SYNTHETIC_ID_FEATURE_FLAG && randomBoolean())
-                            .build(),
+                        templateSettings.build(),
                         new CompressedXContent("""
 
                                 {

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -466,9 +466,6 @@ tests:
 - class: org.elasticsearch.index.mapper.LongSyntheticSourceNativeArrayIntegrationTests
   method: testSynthesizeArrayRandom
   issue: https://github.com/elastic/elasticsearch/issues/138819
-- class: org.elasticsearch.datastreams.TSDBIndexingIT
-  method: testDynamicDimensions
-  issue: https://github.com/elastic/elasticsearch/issues/138815
 - class: org.elasticsearch.xpack.esql.qa.mixed.EsqlClientYamlIT
   issue: https://github.com/elastic/elasticsearch/issues/138796
 

